lib/iris/analysis/trajectory.py

```diff
@@ -237,12 +237,17 @@ def interpolate(cube, sample_points, method=None):
     # Use a cache with _nearest_neighbour_indices_ndcoords()
     cache = {}
 
+    # Cache the linear interpolator
+    scheme = iris.analysis.Linear()
+    coords, points = zip(*sample_points)
+    interpolator = scheme.interpolator(cube, coords)
```

Comment:
These lines need to be protected by `if method in ('linear', None)` - partly because it's wasted effort otherwise, but mostly because they will fail when trying to do "nearest-neighbour interpolation over multi-dimensional coordinates.

NB. The docstring for `iris.analysis.trajectory.interpolate` is a little ambiguous about the default value of `method`, but it turns out that the default is not just simply "linear", but switches to "nearest" when using multi-dimensional coordinates. :-1: I definitely don't propose to change that behaviour in _this_ PR, but it would be nice to clarify the docstring (again, possibly in a separate PR - small is beautiful).